ui/src/components/Sidebar/Sidebar.tsx

```diff
@@ -53,6 +58,22 @@ export default function SideBar({
     });
   };
 
+  const checkFromValidity = (value: string) => {
```

Comment:
Make these functions DRY by combining them into a single function that accepts a callback function, or you could refactor with the suggestion in the next comment

Reply:
This function does check the validity of the timestamp, but it also calls a setState function. The name of the function could be better.

Or, you could make this function **only** check the validity of the timestamp. Rename the function `isTimestampValid` and have it return a boolean.

Move the logic of setting state to the `onClick` handler functions in the TextField components:
```jsx
onChange={(e) => {
  setFromTimestamp(e.target.value);
  if (isValidTimestamp(e.target.value)) setValidFromTimestamp(e.target.value):
}
```